spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/mapping/SpannerMappingContext.java

```diff
@@ -91,9 +95,15 @@ public FieldNamingStrategy getFieldNamingStrategy() {
 
   protected <T> SpannerPersistentEntityImpl<T> constructPersistentEntity(
       TypeInformation<T> typeInformation) {
-    return new SpannerPersistentEntityImpl<>(typeInformation, this,
-        this.applicationContext == null ? new ConverterAwareMappingSpannerEntityProcessor(this)
-            : this.applicationContext.getBean(SpannerEntityProcessor.class));
+    SpannerEntityProcessor processor;
+    if (this.applicationContext == null) {
+      processor = new ConverterAwareMappingSpannerEntityProcessor(this);
+    } else if (!this.applicationContext.containsBean("spannerConverter")) {
+      processor = new ConverterAwareMappingSpannerEntityProcessor(this);
```

Comment:
These can be collapsed into `if (A || B)` ingle branch, since the action is the same